dev-requirements.txt

```diff
@@ -10,3 +10,4 @@ sphinxcontrib_programoutput
 sphinx_autodoc_typehints
 matplotlib
 scipy
+pre-commit
```

Comment:
Are black and the other linters not needed?

Reply:
No, `pre-commit` makes its own separate environment to make sure no rogue configs and similar may interfere.